src/utils/queue.rs

```diff
@@ -13,5 +33,169 @@ pub fn add_crate_to_queue(
         "INSERT INTO queue (name, version, priority) VALUES ($1, $2, $3)",
         &[&name, &version, &priority],
     )?;
+
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::test::wrapper;
+
+    /// Set all crates that match [`pattern`] to have a certain priority
+    ///
+    /// Note: `pattern` is used in a `LIKE` statement, so it must follow the postgres like syntax
+    ///
+    /// [`pattern`]: https://www.postgresql.org/docs/8.3/functions-matching.html
+    pub fn set_crate_priority(conn: &Connection, pattern: &str, priority: i32) -> Result<()> {
+        conn.query(
+            "INSERT INTO crate_priorities (pattern, priority) VALUES ($1, $2)",
+            &[&pattern, &priority],
+        )?;
+
+        Ok(())
+    }
+
+    /// Remove a pattern from the priority table, returning the priority that it was associated with or `None`
+    /// if nothing was removed
+    pub fn remove_crate_priority(conn: &Connection, pattern: &str) -> Result<Option<i32>> {
```

Comment:
Maybe it would be nice to expose these in the binary by adding a `cratesfyi queue priority` subcommand? That would go in https://github.com/rust-lang/docs.rs/blob/master/src/bin/cratesfyi.rs.

Reply:
Would you prefer I do that now or in a separate PR, I've been meaning to refactor `cratesfyi.rs`

Reply:
This should not be part of a refactor PR, I'd do that now.